api/src/main/java/io/opencensus/trace/propagation/BinaryFormat.java

```diff
@@ -65,23 +65,58 @@
   static final NoopBinaryFormat NOOP_BINARY_FORMAT = new NoopBinaryFormat();
 
   /**
-   * Serializes a {@link SpanContext} using the binary format.
-   *
+   * @deprecated use {@link #toByteArray(SpanContext)}.
    * @param spanContext the {@code SpanContext} to serialize.
    * @return the serialized binary value.
    * @throws NullPointerException if the {@code spanContext} is {@code null}.
    */
-  public abstract byte[] toBinaryValue(SpanContext spanContext);
+  @Deprecated
+  public byte[] toBinaryValue(SpanContext spanContext) {
+    return toByteArray(spanContext);
+  }
 
   /**
-   * Parses the {@link SpanContext} from the binary format.
+   * Serializes a {@link SpanContext} using the binary format into a byte array.
    *
+   * @param spanContext the {@code SpanContext} to serialize.
+   * @return the serialized binary value.
+   * @throws NullPointerException if the {@code spanContext} is {@code null}.
+   */
+  public byte[] toByteArray(SpanContext spanContext) {
+    return toBinaryValue(spanContext);
+  }
+
+  /**
+   * @deprecated use {@link #fromByteArray(byte[])}.
    * @param bytes a binary encoded buffer from which the {@code SpanContext} will be parsed.
    * @return the parsed {@code SpanContext}.
    * @throws NullPointerException if the {@code input} is {@code null}.
    * @throws ParseException if the version is not supported or the input is invalid
    */
-  public abstract SpanContext fromBinaryValue(byte[] bytes) throws ParseException;
+  @Deprecated
+  public SpanContext fromBinaryValue(byte[] bytes) throws ParseException {
+    try {
+      return fromByteArray(bytes);
+    } catch (SpanContextParseException e) {
+      throw new ParseException(e.toString(), 0);
+    }
+  }
+
+  /**
+   * Parses the {@link SpanContext} using the binary format from a byte array.
+   *
+   * @param bytes a binary encoded buffer from which the {@code SpanContext} will be parsed.
+   * @return the parsed {@code SpanContext}.
+   * @throws NullPointerException if the {@code input} is {@code null}.
+   * @throws SpanContextParseException if the version is not supported or the input is invalid
+   */
+  public SpanContext fromByteArray(byte[] bytes) throws SpanContextParseException {
+    try {
+      return fromBinaryValue(bytes);
+    } catch (ParseException e) {
```

Comment:
Since the result type is checked exception should we also wrap runtime exceptions or let them propagate?

Reply:
we do throw only one type of runtime exception which is NullPointerException if the buffer is null.